src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { Todo } from '../types/Todo';
+import { Filters } from '../types/Filters';
+
+type Props = {
+  currentFilter: Filters;
+  setCurrentFilter: Dispatch<SetStateAction<Filters>>;
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+};
+
+export const TodoFooter: React.FC<Props> = ({
+  todos,
+  setTodos,
+  currentFilter,
+  setCurrentFilter,
+}) => {
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const handleClearCompleted = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
+  const filters = [
+    { name: 'All', href: '#/', filter: Filters.All, dataCy: 'FilterLinkAll' },
+    {
+      name: 'Active',
+      href: '#/active',
+      filter: Filters.Active,
+      dataCy: 'FilterLinkActive',
+    },
+    {
+      name: 'Completed',
+      href: '#/completed',
+      filter: Filters.Completed,
+      dataCy: 'FilterLinkCompleted',
+    },
+  ];
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {activeTodos.length} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        {filters.map(({ name, href, filter, dataCy }) => (
+          <a
+            key={filter}
+            href={href}
+            className={`filter__link ${currentFilter === filter ? 'selected' : ''}`}
```

Comment:
To set classes conditionally use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library